### PR TITLE
Add linter that targets only compositions folder

### PR DESCRIPTION
**Pull request by barney-s**
### Change description

For some reason the global linter was not catching any errors in experiments. 

Add a github actions for experiments/compositions:
- go linter

### CODE REVIEW

**Comment by barney-s** [comment]
Linter errors surfaced:

```
Running [/home/runner/golangci-lint-1.59.1-linux-amd64/golangci-lint run --out-format=github-actions --path-prefix=./experiments/compositions/composition] in [/home/runner/work/k8s-config-connector/k8s-config-connector/experiments/compositions/composition] ...
  Error: 180-221 lines are duplicate of `expanders/getter/main_test.go:266-307` (dupl)
  Error: 266-307 lines are duplicate of `expanders/getter/main_test.go:180-221` (dupl)
  Error: Error return value of `conn.Close` is not checked (errcheck)
  Error: Error return value of `os.RemoveAll` is not checked (errcheck)
  Error: Error return value of `os.RemoveAll` is not checked (errcheck)
  Error: Error return value of `os.Remove` is not checked (errcheck)
  Error: Error return value of `clusterConfigFile.Close` is not checked (errcheck)
  Error: Error return value of `os.Remove` is not checked (errcheck)
  Error: Error return value of `kubeConfigFile.Close` is not checked (errcheck)
  Error: Error return value of `conn.Close` is not checked (errcheck)
  Error: Error return value of `item.SetNamespace` is not checked (errcheck)
  Error: Error return value of `os.MkdirAll` is not checked (errcheck)
  Error: Error return value of `os.RemoveAll` is not checked (errcheck)
  Error: Error return value of `podLogs.Close` is not checked (errcheck)
  Error: Error return value of `file.Close` is not checked (errcheck)
  Error: Error return value of `mgr.Start` is not checked (errcheck)
  Error: string `{"key": "val"}` has 7 occurrences, make it a constant (goconst)
  Error: string `has no attribute 'zone'` has 3 occurrences, make it a constant (goconst)
  Error: string `Complete` has 3 occurrences, make it a constant (goconst)
  Error: string `Failed` has 3 occurrences, make it a constant (goconst)
  Error: cyclomatic complexity 37 of func `(*ExpanderReconciler).Reconcile` is high (> 30) (gocyclo)
  Error: File is not `gofmt`-ed with `-s` (gofmt)
  Error: File is not `gofmt`-ed with `-s` (gofmt)
  Error: File is not `gofmt`-ed with `-s` (gofmt)
  Error: line is 140 characters (lll)
  Error: line is 172 characters (lll)
  Error: line is 156 characters (lll)
  Error: line is [34](https://github.com/GoogleCloudPlatform/k8s-config-connector/actions/runs/10378767545/job/28735645655?pr=2493#step:4:36)7 characters (lll)
  Error: line is 346 characters (lll)
  Error: line is 217 characters (lll)
  Error: line is 122 characters (lll)
  Error: line is 123 characters (lll)
  Error: line is 147 characters (lll)
  Error: line is 148 characters (lll)
  Error: line is 152 characters (lll)
  Error: line is 1[35](https://github.com/GoogleCloudPlatform/k8s-config-connector/actions/runs/10378767545/job/28735645655?pr=2493#step:4:37) characters (lll)
  Error: line is 127 characters (lll)
  Error: line is 124 characters (lll)
  Error: line is 142 characters (lll)
  Error: line is 141 characters (lll)
  Error: line is 183 characters (lll)
  Error: line is 177 characters (lll)
  Error: line is 176 characters (lll)
  Error: line is 123 characters (lll)
  Error: line is 147 characters (lll)
  Error: line is 174 characters (lll)
  Error: line is 128 characters (lll)
  Error: line is 127 characters (lll)
  Error: line is 127 characters (lll)
  Error: line is 128 characters (lll)
  Error: line is 134 characters (lll)
  Error: line is 130 characters (lll)
  Error: line is 122 characters (lll)
  Error: line is 122 characters (lll)
  Error: line is 121 characters (lll)
  Error: line is 121 characters (lll)
  Error: line is 123 characters (lll)
  Error: line is 126 characters (lll)
  Error: line is 126 characters (lll)
  Error: line is 135 characters (lll)
  Error: `succesfully` is a misspelling of `successfully` (misspell)
  Error: `succesfully` is a misspelling of `successfully` (misspell)
  Error: `succesfully` is a misspelling of `successfully` (misspell)
  Error: `succeded` is a misspelling of `succeeded` (misspell)
  Error: Consider pre-allocating `reqs` (prealloc)
  Error: unnecessary conversion (unconvert)
  Error: (*expander).Validate - result 1 (error) is always nil (unparam)
  Error: `getFrequency` - `duration` always receives `opDuration` (`2000000000`) (unparam)
  Error: `(*ExpanderVersionReconciler).processExpanderVersion` - `ctx` is unused (unparam)
  Error: (*ExpanderVersionReconciler).processExpanderVersion - result 0 (error) is always nil (unparam)
  Error: var `planGVK` is unused (unused)
  Error: field `namespace` is unused (unused)
  Error: S1006: should use for {} instead of for true {} (gosimple)
  Error: S1006: should use for {} instead of for true {} (gosimple)
  Error: ineffectual assignment to err (ineffassign)
  Error: ineffectual assignment to err (ineffassign)
  Error: ineffectual assignment to err (ineffassign)
  Error: SA[40](https://github.com/GoogleCloudPlatform/k8s-config-connector/actions/runs/10378767545/job/28735645655?pr=2493#step:4:42)09: argument v is overwritten before first use (staticcheck)
  Error: SA4009(related information): assignment to v (staticcheck)
  Error: SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x.  (staticcheck)
  Error: SA1019: "io/ioutil" has been deprecated since Go 1.19: As of Go 1.16, the same functionality is now provided by package [io] or package [os], and those implementations should be preferred in new code. See the specific function documentation for details.  (staticcheck)
  Error: SA1019: "io/ioutil" has been deprecated since Go 1.19: As of Go 1.16, the same functionality is now provided by package [io] or package [os], and those implementations should be preferred in new code. See the specific function documentation for details.  (staticcheck)
  Error: SA1019: rand.Seed has been deprecated since Go 1.20 and an alternative has been available since Go 1.0: As of Go 1.20 there is no reason to call Seed with a random value. Programs that call Seed with a known value to get a specific sequence of results should use New(NewSource(seed)) to obtain a local random generator.  (staticcheck)
  Error: SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x.  (staticcheck)
  Error: SA1019: rand.Seed has been deprecated since Go 1.20 and an alternative has been available since Go 1.0: As of Go 1.20 there is no reason to call Seed with a random value. Programs that call Seed with a known value to get a specific sequence of results should use New(NewSource(seed)) to obtain a local random generator.  (staticcheck)
  Error: SA1019: grpc.Dial is deprecated: use NewClient instead.  Will be supported throughout 1.x.  (staticcheck)
  Error: SA1019: rand.Seed has been deprecated since Go 1.20 and an alternative has been available since Go 1.0: As of Go 1.20 there is no reason to call Seed with a random value. Programs that call Seed with a known value to get a specific sequence of results should use New(NewSource(seed)) to obtain a local random generator.  (staticcheck)
  Error: SA9004: only the first constant in this group has an explicit type (staticcheck)
  Error: SA1004: sleeping for 2 nanoseconds is probably a bug; be explicit if it isn't (staticcheck)
  Error: SA1004: sleeping for 2 nanoseconds is probably a bug; be explicit if it isn't (staticcheck)
  
  level=warning msg="[config_reader] The output format `github-actions` is deprecated, please use `colored-line-number`"
  level=error msg="Timeout exceeded: try increasing it by passing --timeout option"
  
  Error: golangci-lint exit with code 4
  Ran golangci-lint in 77[60](https://github.com/GoogleCloudPlatform/k8s-config-connector/actions/runs/10378767545/job/28735645655?pr=2493#step:4:62)8ms
```

File: experiments/compositions/composition/expanders/getter/main_test.go [hunk 1]
```diff
@@ -177,6 +177,25 @@ func TestEvaluateEmptyFacade(t *testing.T) {
...
+	if err != nil {
+		t.Fatalf("expected no error. got %v", err)
+	}
+
+	if r.Status != pb.Status_EVALUATE_WAIT {
+		t.Fatalf("\nexpected: EVALUATE_WAIT \n got: %s", r)
+	}
+
+	if !strings.Contains(r.Error.Message, expectedErrorString) {
+		t.Fatalf("expected error contains: %s \n got: %s", expectedErrorString, r)
```
Comment (justinsb) [inline experiments/compositions/composition/expanders/getter/main_test.go:195]: Nit: I tend to find that %q is easier to read here (and shows tabs vs spaces etc)

**Review by justinsb — APPROVED** [review]

File: experiments/compositions/composition/expanders/getter/main_test.go (hunk 1 above)
Comment (barney-s) [reply experiments/compositions/composition/expanders/getter/main_test.go:195]: ack.

**Comment by justinsb** [comment]
For the "must check errors" you might want to create a helper for the tests like this:

```
Must(t *testing.T, err error) {
  if err != nil {
    t.Errorf("unexpected error: %v", err)
  }
}
```

2 downsides:

1) You can't use t.Fatalf because it's in a defer block, and I think another vet rule will yell at you if you try
2) The error message is going to be pretty obscure

Because of (2) I would suggest something like:  `MustClose(t *testing.T, r io.Closeable)`.  But then you need more helper methods, so /shrug

**Pull request merged**